numpy/lib/function_base.py

```diff
@@ -305,11 +305,16 @@ def average(a, axis=None, weights=None, returned=False):
     Returns
     -------
     average, [sum_of_weights] : array_type or double
-        Return the average along the specified axis. When returned is `True`,
+        Return the average along the specified axis. When `returned` is `True`,
         return a tuple with the average as the first element and the sum
-        of the weights as the second element. The return type is `Float`
-        if `a` is of integer type, otherwise it is of the same type as `a`.
-        `sum_of_weights` is of the same type as `average`.
+        of the weights as the second element. `sum_of_weights` is of the
+        same type as `average`. The result dtype follows a genereal pattern.
+        If `weights` is None, the result dtype will be that of `a` , or `float64`
+        if `a` is integral. Otherwise, if `weights` is not None and `a` is non
+        integral, the result type will be the type of lowest precision capable of
+        representing values of both `a` and `weights` but if `a` happens to be
```

Comment:
That all seems correct but it's a pretty long sentence. Maybe you can replace the first "but" by a period.